WordPress/gutenberg
# Remove not needed aria-hidden attribute in navigation block experiment

## What?
As reported [here](https://wordpress.slack.com/archives/C02RP4X03/p1685133083579679?thread_ts=1684403764.422519&cid=C02RP4X03): In the version of the navigation block using the Interactivity API, in some cases we are defining `aria-hidden: true` even for the desktop version, which is not correct. This pull request aims to solve that.

## Why?
It causes issues with screen readers in the desktop version, when the elements are shown but `aria-hidden` is true.

## How?
~~I added a selector that reads:~~
* ~~If the menu is open -> `aria-hidden: false`.~~
* ~~If the menu is not open -> It checks if the "Open menu" is visible. If it is, `aria-hidden: true`. If it isn't `aria-hidden: false`.~~

~~I couldn't find any other way of checking this because, if I am not mistaken, it is handled with CSS.~~

I finally removed the aria-hidden attribute directly because it seems that it shouldn't be used in this use case, where the element is hidden using `display: none`, according to [MDN docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden#:~:text=aria%2Dhidden%3D%22true%22%20should%20not%20be%20added,the%20element%27s%20ancestor%20is%20hidden%20with%20visibility%3A%20hidden): _"aria-hidden="true" should not be added when the element or the element's ancestor is hidden with [display: none](https://developer.mozilla.org/en-US/docs/Web/CSS/display)"_.

## Testing Instructions

1. Open a site with a Navigation block that is responsive: It is wrapped in mobile but not in desktop.
2. In the Gutenberg Experiments page, activate the "Test the core blocks using the Interactivity API" option.
3. In the mobile version, check that `aria-hidden` inside the `wp-block-navigation__responsive-container` is true when the menu is wrapped, and false when the menu opens.
4. In the desktop version, check that `aria-hidden` is always false.

## Screenshots or screencast

**_Desktop version_**
![Screenshot 2023-05-29 at 11 15 55](https://github.com/WordPress/gutenberg/assets/34552881/1d338cf1-e751-4332-8df3-dbad7b162be8)


**_Mobile version_**
![Screenshot 2023-05-29 at 11 16 45](https://github.com/WordPress/gutenberg/assets/34552881/a625e9fe-d106-493a-9565-40fd24d13321)
